### ISSUE TITLE
Absolute paths do not work with rego.Load on Windows

### ISSUE THREAD

**Issue by lcarva** [open-policy-agent/opa#4521]
## Short description 
When using `rego.Load` with an absolute path on Windows, the following error is returned when `Rego.PrepareForEval` is called:
```
1 error occurred during loading: CreateFile \Users\RUNNER~1\AppData\Local\Temp\TestRegoPolicyLoadAbsolutePath3698015745\001\policy.rego: The system cannot find the path specified.
```

The issue does not occur if a relative path is used. The issue does not work on Linux or OSX.

## Steps To Reproduce

```go
policy := `
	package signature

	allow {
		input.predicateType == "https://slsa.dev/provenance/v0.2"
	}
`
// This alternative line works:
// policyFile := "policy.rego"
policyFile := filepath.Join(t.TempDir(), "policy.rego")
if err := os.WriteFile(policyFile, []byte(policy), 0644); err != nil {
	t.Fatal(err)
}
r := rego.New(rego.Query("data.signature.allow"), rego.Load([]string{policyFile}, nil))

ctx := context.Background()

// This fails on Windows Server 2022.
_, err := r.PrepareForEval(ctx)
if err != nil {
	t.Fatal(err)
}
```

## Expected behavior
`PrepareForEval` should not return an error.

## Additional context
I created a small git [repo](https://github.com/lcarva/writefile-test/blob/a5a28f2de041adfeb9f48f7d3662528a06bd5975/main_test.go) to reproduce this issue. The test `TestRegoPolicyLoadAbsolutePath` fails, while the test `TestRegoPolicyLoadRelativePath` passes. The GitHub actions were useful in running the tests across different Operating Systems. I'm not sure how long the results will last, but here's a [link](https://github.com/lcarva/writefile-test/runs/5755552826?check_suite_focus=true) to them.

**Comment by anderseknert** [comment]
Hey @lcarva! Sorry for not acknowledging this before, but that definitely seems like a bug. If anyone with a Windows machine would consider taking a look at this, that would be much appreciated!

**Comment by lcarva** [comment]
@anderseknert, no worries! FWIW, it's possible to reproduce the issue with just GitHub actions as mentioned in the description. It's definitely not as easy to debug as a local system, but a potential way forward.

**Comment by lcarva** [comment]
Still failing on opa v0.40.0 with the same error.

**Comment by anderseknert** [comment]
Yes, no changes here, @lcarva. The stalebot warning does not mean it's being closed/removed, just that there's not been any activity for some time.

**Comment by simar7** [comment]
Still seems to be the case on v0.54.0.

**Comment by mariuspodean** [comment]
I think the issue here spans over multiple areas were files need to be loaded; same error when trying to use certificates (`--tls-cert-file`, `--tls-private-key-file`, `--tls-ca-cert-file`) from an absolute path (v0.68)

**Comment by mariuspodean** [comment]
Possibly similar > https://github.com/open-policy-agent/opa/issues/6910

**Comment by alexrohozneanu** [comment]
In the scenario where we are running the opa executable on one drive and loading the files from another drive then files cannot be found.
However, if the opa executable and the files to be loaded are in the same drive, everything is working.

```
opa run \
        -w --server \
        -b F:/opa/bundle.tar.gz \
        --log-level debug \
        --tls-cert-file C:/OPACerts/newcerts/SERVER.pem \
        --tls-private-key-file C:/OPACerts/keys/SERVER.key  \
        --tls-ca-cert-file C:/OPACerts/newcerts/CA.pem \
        --addr https://0.0.0.0:8181/ \
        --log-format=json-pretty \
        --set=decision_logs.console=true \
        --authentication=tls \
        --authorization=basic
```

It seems that in the following function we are loosing information about the prefix:

https://github.com/open-policy-agent/opa/blob/69cd3886ea2fd6ec167b5c6d935724303a794c48/internal/pathwatcher/utils.go#L100

On windows, if we comment out that line, files are loaded correctly.

@ashutosh-narkar Can you give us more details about why is the prefix ignored, since all the paths in windows have the following format {partition_letter}:/{actual_path}? e.g. C:/Folder/file

**Comment by ashutosh-narkar** [comment]
IIRC it may be assumed the watchers are set for path in the current directory. If you think there is scope for improvement here, feel free to submit a PR.

**Comment by mariuspodean** [comment]
> IIRC it may be assumed the watchers are set for path in the current directory. If you think there is scope for improvement here, feel free to submit a PR.

I think there is, indeed, a need for a fix here; whether you opt for watches, or not, this is still called and basically cannot start the server on a Windows machine
